### PR TITLE
socket_vmnet 1.1.0

### PR DESCRIPTION
**Pull request by AkihiroSuda**
`socket_vmnet` (https://github.com/lima-vm/socket_vmnet) provides `vmnet.framework` support for rootless QEMU.

`socket_vmnet` has been already an optional dependency for Lima and Minikube:
- Lima: https://github.com/lima-vm/lima/blob/master/docs/network.md#managed-vmnet-networks-192168105024
- Minikube: https://minikube.sigs.k8s.io/docs/drivers/qemu/

The `socket_vmnet` daemon works as the root, so it has to be installed into a directory that is not writable by non-root users.
(Otherwise a non-root user can gain the root by replacing the `socket_vmnet` daemon binary)
For this reason, this PR is opened to homebrew-cask, not homebrew-core.

- - -
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [X] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
🙏  **Exception request**: (see `audit for socket_vmnet: failed` below)
- [X] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
🙏 **Exception request**: the underscore is included in the cask token because the underscore is the part of the project name
- [X] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [X] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
```
audit for socket_vmnet: failed
 - cask token underscores should be replaced by hyphens
 - cask token should only contain lowercase alphanumeric characters and hyphens
 - GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 3 problems in 1 cask detected
```
🙏 **Exception requests**:
  - The underscore in the cask token is the part of the project name
  - The number of the star of https://github.com/lima-vm/socket_vmnet is not enough, but `socket_vmnet` has been already an optional dependency for https://github.com/lima-vm/lima (9.8k stars) and https://github.com/kubernetes/minikube (25.2k stars)

- [X] `brew install --cask <cask>` worked successfully.
- [X] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/socket_vmnet.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    # The daemon (/opt/socket_vmnet/bin/socket_vmnet) run as the root.
+    # So, its executable path must NOT be writable by non-root.
+    set_ownership("/opt/socket_vmnet", user: "root", group: "wheel")
+    set_ownership("/Library/LaunchDaemons/io.github.lima-vm.socket_vmnet.plist", user: "root", group: "wheel")
+    # Run spctl --add to avoid the "cannot be opened because the developer cannot be verified" error
+    # https://github.com/alacritty/alacritty/issues/4673#issuecomment-771291615
+    system_command "/usr/sbin/spctl",
+                   args: ["--add",
+                          "/opt/socket_vmnet/bin/socket_vmnet",
+                          "/opt/socket_vmnet/bin/socket_vmnet_client"]
```
Comment (AkihiroSuda) [inline Casks/socket_vmnet.rb:33]: (CI hanging here)
```
/usr/sbin/spctl --add /opt/socket_vmnet/bin/socket_vmnet /opt/socket_vmnet/bin/socket_vmnet_client
Error: The action has timed out.
```

https://github.com/Homebrew/homebrew-cask/actions/runs/3521371144/jobs/5903168970

**Review by SMillerDev — CHANGES_REQUESTED** [review]
That's waaaaay too many custom artifacts and caveats for my taste. It also seems to be an open source commandline tool so it's better suited as a formula.

**Comment by AkihiroSuda** [comment]
> That's nice, but it's still essentially writing a completely custom installer in the preflight block, including disabling Homebrew security measures.

The preflight block is just for removing the `com.apple.quarantine` xattr and installing launchd plist.
The `com.apple.quarantine` xattr is removed (equates to clicking the "accept" button in the "Security" pane of System Preferences), but the sha256sum of the binary is still verified by Homebrew, so I don't see a security issue.

> Which is not a designed feature of Homebrew.

What would be the right way to distribute a daemon program that has to run as the root?

**Comment by AkihiroSuda** [comment]
Draft for non-cask PR: https://github.com/AkihiroSuda/homebrew-core/blob/socket_vmnet/Formula/socket_vmnet.rb

@SMillerDev Does this look better?

**Comment by p-linnane** [comment]
@AkihiroSuda Please open a PR in `homebrew-core` to get feedback. We will not be able to accept this in `homebrew-cask`.

**Comment by AkihiroSuda** [comment]
@SMillerDev @p-linnane Could you review the `homebrew-core` PR?
- https://github.com/Homebrew/homebrew-core/pull/116501

**Pull request closed without merging**